app/routes/public/sessions/list.js

```diff
@@ -16,169 +16,63 @@ export default Route.extend({
+            name : 'event',
+            op   : 'has',
+            val  : {
+              name : 'identifier',
+              op   : 'eq',
+              val  : eventDetails.id
+            }
+          },
           {
-            and: [
+            or: [
               {
-                name : 'event',
-                op   : 'has',
-                val  : {
-                  name : 'identifier',
-                  op   : 'eq',
-                  val  : eventDetails.id
-                }
+                name : 'state',
+                op   : 'eq',
+                val  : 'confirmed'
               },
               {
-                name : 'starts-at',
-                op   : 'ge',
-                val  : moment().startOf('week').toISOString()
-              },
-              {
-                name : 'starts-at',
-                op   : 'lt',
-                val  : moment().endOf('week').toISOString()
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
+                name : 'state',
+                op   : 'eq',
+                val  : 'accepted'
               }
             ]
           }
         ]
-      });
-    } else if (params.session_status === 'month') {
-      sessions = await this.store.query('session', {
-        filter: [
+      }
+    ];
+
+    if (params.session_status !== 'all') {
+
+      filterOptions.push({
+        and: [
           {
-            and: [
-              {
-                name : 'event',
-                op   : 'has',
-                val  : {
-                  name : 'identifier',
-                  op   : 'eq',
-                  val  : eventDetails.id
-                }
-              },
-              {
-                name : 'starts-at',
-                op   : 'ge',
-                val  : moment().startOf('month').toISOString()
-              },
-              {
-                name : 'starts-at',
-                op   : 'lt',
-                val  : moment().add('month').toISOString()
-              },
-              {
-                or: [
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'confirmed'
-                  },
-                  {
-                    name : 'state',
-                    op   : 'eq',
-                    val  : 'accepted'
-                  }
-                ]
-              }
-            ]
-          }
-        ]
-      });
-    } else {
-      sessions = await this.store.query('session', {
-        filter: [
+            name : 'starts-at',
+            op   : 'ge',
+            val  : moment().startOf(`${params.session_status}`).toISOString()
```

Comment:
For 'day' it was 'today' I suppose

Reply:
There will be an extra conditional for today then.

Reply:
```js
const period = params.session_status === 'today' ? 'day' : params.session_status;
```

Reply:
> ```js
> const period = params.session_status === 'today' ? 'day' : params.session_status;
> ```

Thanks, Done!